reference-library/BeforeSwapDelta.md

```diff
@@ -0,0 +1,218 @@
+# BeforeSwapDelta
+
+`BeforeSwapDelta` is a custom type used in Uniswap V4 hook contracts to represent balance changes during swap operations. It is specifically designed to handle the return value of the `beforeSwap` hook and to be compatible with the `afterSwap` hook.
+
+Before explaining `BeforeSwapDelta` in detail, it is worth noting that in the context of Uniswap V4 swaps:
+
+- The "specified" token is the one for which the user specifies an exact input or output amount.
+- The "unspecified" token is the counterpart in the swap, whose amount is determined by the pool's pricing mechanism.
+
+## Purpose
+
+The main purpose of `BeforeSwapDelta` is to efficiently encode and decode balance changes for both specified and unspecified tokens in a single `int256` value. This compact representation allows for gas-efficient operations and seamless integration with Uniswap V4's hook system.
+
+`BeforeSwapDelta` is essential for:
+
+- Allowing hooks to modify swap parameters or override default swap behavior
+- Enabling information passing between `beforeSwap` and `afterSwap` hooks
```

Comment:
```suggestion
```

a bit ambiguous and initially reads as an arbitrary bytes handler (similar to hookData), so we can exclude this